### PR TITLE
Adds new dog breed + two evolved dog breeds

### PR DESCRIPTION
**Pull request by f-rovers**
#### Summary
Content "Adds a new dog breed /w evolution + evolution for rottweiler"

#### Purpose of change
I've always found it odd that cougars could mutate, but not large dogs, who're in the same weight class
The evolved dogs also help out the 'pet owner' playstyle by giving them something actually decently powerful to work with.

#### Describe the solution
Adds the greater swiss mountain dog, its evolution the greatest swiss mountain dog and their puppies.
Greater swiss mountain dogs are slightly weaker, but bulkier rottweilers in terms of stats
Greatest swiss mountain dogs are intended for pulling carts or as a slower alternative to horses. However, you can ride these into battle. Two of them can also pull the gig carriage at 25 km/h on roads, but struggle off-road. In combat, they aren't all that impressive. They can take a good amount of hits, but their moral isn't the best and they end up fleeing at around half hp.

Also adds the evolution to the rottweiler, the gottweiler. These are a lot more dangerous, both to players and to monsters once tamed. They have some bash armour, decent cut armour but lack piercing and ballistic resistance. They also have a regen factor, which makes them great attack dogs or hard to kill for early game players. ~~To prevent early taming, the gottweiler has a "PLAYER_CLOSE" anger trigger. Its aggression has been hand-tested to allow only a survivor with Animal Kinship to approach and tame them directly. Others will have to wound it enough to break its moral first.~~ On second thought, feeding a several hundred kilo predator a handful of kibble and having it serve you unconditionally was deemed silly. The puppies can still be tamed

#### Describe alternatives you've considered
Nothing really, I just wanted to start working on evolved dogs

#### Testing
Booted game to ensure there were no errors

Tested morale multiple times. 
Eventually got to a morale of 30 where it fought back long enough to be threatning (till about 50% HP)
With 30 it also fought the player when puppies were nearby, fought to the death while puppies were nearby.
Gottweiler ended up with slightly higher base morale (40) to both make it a better combat pet and make it harder to kill/tame

-Combat test 8/8/8/8 player vs greatest swiss-
Unarmed-no skill = player death. <80% hp remaining on greatest swiss
Switched to the standard combat test suite
Evacuee w/ knife spear = victory, ~20% damage on three limbs.
Evacuee w/ pipe mace = death, ~70% hp remaining on greatest swiss
Evacuee w/ makeshift machete = victory, ~30% torso health, 20-30% damage on legs and head.
Evacuee w/ glock duty pistol (other side of the shelter aka 12 tiles) = victory, no damage, does take roughly a dozen bullets

Scavenger w/ fireaxe = victory, no damage. Morale broke in two hits.
Scavenger w/ shillelagh = victory, greatest swiss only gets one or two hits in, morale broke in 4-5 hits.
Scavenger w/ combat knife = victory, greatest swiss only gets one or two hits in, morale broke in 4-5 hits.
Scavenger w/ AR15 = victory, no damage. Morale broke in two hits. Took 5 bullets.

Survivor w/ broadsword = victory, no damage, morale broke in two hits
Survivor w/ lucrene hammer = victory, no damage, morale broke on the first hit
Survivor w/ nodachi = victory, no damage, morale broke in two hits.
Survivor w/ mace = victory, no damage, morale broke in 2-3 hits
Survivor w/ rapier = victory, no damage, morale broke in two hits
Survivor w/ pike = victory, no damage, morale broke in two hits. Did close the 2-tile gap in one attack.
Survivor w/ barrett = victory, no damage, morale broke on the first hit. Took 2 bullets.

-Combat test 8/8/8/8 player vs gottweiler-
Evacuee unarmed = death, no damage dealt.
Evacuee w/ knife spear = death, managed to build up heavy beelding.
Evacuee w/ pipe mace = death, no lasting damage dealt.
Evacuee w/ makeshift machete = death, no lasting damage dealt.
Evacuee w/ glock duty pistol = survived. Managed to break morale after 8-10 bullets.

Scavenger w/ fireaxe = Victory, ~10% damage taken to head, broke morale after 4 hits.
Scavenger w/ shillelagh = Victory, ~20% damage taken to head and leg, it managed to shatter a plate, morale broke after 8 hits
Scavenger w/ combat knife = Victory, ~30% damage taken to one arm, morale broke in 6 hits
Scavenger w/ AR15 = Victory, no damage taken, morale broke in 3 hits, kill took 8 bullets

Survivor w/ broadsword = victory, no damage, morale broke in two hits
Survivor w/ lucrene hammer = victory, no damage, morale broke in two hit
Survivor w/ nodachi = victory, no damage, morale broke in two hits.
Survivor w/ mace = victory, no damage, morale broke in five hits
Survivor w/ rapier = victory, no damage, morale broke in two hits
Survivor w/ pike = victory, no damage, morale broke in three hits.
Survivor w/ barrett = victory, no damage, morale broke on the first hit. Took 3 bullets.

Also tested both dogs as combat pets. 
Greater Swiss will die if pitched against more than 2 or 3 common zombies
Gottweiler can handle groups of common infected, especially if the player is fighting alongside them. If not, their morale drops fairly quickly and they'll flee.
Gottweiler was also vulnerable to acid/electric zombies

#### Additional context
I am open to suggestions for the evolution of other dog breeds, right now I've stuck to the two I had an idea for.
(Cartdog and guard/combat dog respectively)
Edit: Moved evolved dogs to mutant_mammal, together with the tusked moose

### CODE REVIEW

**Comment by Karol1223** [comment]
Honestly, I'm not a fan of cow-sized predators being tamable with some dog food. That does not seem very appropriate. If they do mutate from regular versions of the respective dog breeds (like they should) then they should be obtainable anyway - by an already tamed dog mutating - but taming adult dogs the size of a cow, which have at that point become incomparably higher in the food chain than a human, isn't really appropriate. A dog this size would likely just rather take you as food alongside the dogfood you're offering...

**Comment by f-rovers** [comment]
That sounds fair honestly. I haven't been able to test if they mutate properly, though I assume they do as I'm using the same code as is used for the moose => great tusked moose.

I'll remove the ability to tame them while I also correct the single/plural form of greater swiss

**Comment by f-rovers** [comment]
Mabye "Tamable wildlife" can reintroduce taming them early

File: data/json/monsters/mammal.json [hunk 1]
```diff
@@ -1746,6 +1746,248 @@
...
+      "feed": "The %s seems to like you!  It lets you pat its head and seems friendly.",
+      "pet": "The %s happily wags its tail while you pat their head."
+    },
+    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PET_MOUNTABLE", "CANPLAY", "PATH_AVOID_DANGER_1", "WARM" ]
+  },
+  {
+    "id": "mon_bigdog_gottweiler_pup",
+    "type": "MONSTER",
+    "name": { "str": "gottweiler puppy", "str_pl": "gottweiler puppies" },
+    "description": "An adorable, defenseless Gottweiler puppy.  Much safer to tame than an adult dog, in spite of its already impressive size.",
```
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json:1799]:
```suggestion
    "description": "An adorable, defenseless gottweiler puppy.  Much safer to tame than an adult dog, in spite of its already impressive size.",
```

**Comment by f-rovers** [comment]
On second thought, I agree, I will leave the ability to tame the puppies, who're in a much more appropriate size range.
Do you think a "PLAYER_CLOSE" anger flag for the Greatest Swiss would be a good idea in that case? Or is the "PLAYER_NEAR_BABY" flag enough?

They should be able to evolve properly, though I have not confirmed this. I used the same code as the evolution for moose => great tusked moose.

File: data/json/monsters/mammal.json [hunk 2]
```diff
@@ -1746,6 +1746,248 @@
...
+      "pet": "The %s happily wags its tail while you pat their head."
+    },
+    "flags": [ "ANIMAL", "DOGFOOD", "CANPLAY", "HEARS", "KEENNOSE", "PATH_AVOID_DANGER_1", "SEES", "SMELLS", "WARM" ]
+  },
+  {
+    "id": "mon_dog_gswiss",
+    "copy-from": "mon_dog",
+    "type": "MONSTER",
+    "name": { "str": "greater swiss", "str_pl": "greater swiss" },
+    "description": "The biggest of four breeds of Swiss mountain dogs.  The Greater Swiss mountain dog was originally meant to pull carts, herd cattle and guard the homestead.  This one seems feral, but perhaps you could try taming it?",
```
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json:1839]:
```suggestion
    "name": { "str": "Greater Swiss", "str_pl": "Greater Swisses" },
    "description": "The biggest of four breeds of Swiss mountain dogs.  The Greater Swiss Mountain Dog was originally meant to pull carts, herd cattle and guard the homestead.  This one seems feral, but perhaps you could try taming it?",
```

File: data/json/monsters/mammal.json [hunk 3]
```diff
@@ -1746,6 +1746,248 @@
...
+      "PATH_AVOID_DANGER_1",
+      "SEES",
+      "SMELLS",
+      "WARM"
+    ]
+  },
+  {
+    "id": "mon_dog_gswiss_pup",
+    "type": "MONSTER",
+    "name": { "str": "greater swiss puppy", "str_pl": "greater swiss puppies" },
```
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json:1877]:
```suggestion
    "name": { "str": "Greater Swiss puppy", "str_pl": "Greater Swiss puppies" },
```

File: data/json/monsters/mammal.json [hunk 4]
```diff
@@ -1746,6 +1746,248 @@
...
+      "food": [ "DOGFOOD" ],
+      "feed": "The %s seems to like you!  It lets you pat its head and seems friendly.",
+      "pet": "The %s happily wags its tail while you pat their head."
+    },
+    "flags": [ "ANIMAL", "DOGFOOD", "CANPLAY", "HEARS", "KEENNOSE", "PATH_AVOID_DANGER_1", "SEES", "SMELLS", "WARM" ]
+  },
+  {
+    "id": "mon_bigdog_greatestswiss",
+    "type": "MONSTER",
+    "name": { "str": "greatest swiss", "str_pl": "greatest swiss" },
```
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json:1914]:
```suggestion
    "name": { "str": "Greatest Swiss", "str_pl": "Greatest Swiss" },
```

File: data/json/monsters/mammal.json [hunk 5]
```diff
@@ -1746,6 +1746,248 @@
...
+      "food": [ "DOGFOOD" ],
+      "feed": "The %s seems to like you!  It lets you pat its head and seems friendly.",
+      "pet": "The %s happily wags its tail while you pat their head."
+    },
+    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PET_MOUNTABLE", "CANPLAY", "PATH_AVOID_DANGER_1", "WARM" ]
+  },
+  {
+    "id": "mon_bigdog_greatestswiss_pup",
+    "type": "MONSTER",
+    "name": { "str": "greatest swiss puppy", "str_pl": "greatest swiss puppies" },
```
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json:1956]:
```suggestion
    "name": { "str": "Greatest Swiss puppy", "str_pl": "Greatest Swiss puppies" },
```

**Review by TheShadowFerret — COMMENTED** [review]
Dog breeds are proper nouns 🐶

File: data/json/monsters/mammal.json [hunk 6]
```diff
@@ -1746,6 +1746,248 @@
     },
     "flags": [ "ANIMAL", "DOGFOOD", "CANPLAY", "HEARS", "KEENNOSE", "PATH_AVOID_DANGER_1", "SEES", "SMELLS", "WARM" ]
   },
+  {
+    "id": "mon_bigdog_gottweiler",
+    "type": "MONSTER",
+    "name": { "str": "gottweiler" },
+    "description": "This rottweiler has grown to the proportions of a cow.  Its short fur is riddled with scars and the hide underneath has grown thick and leathery.",
```
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json:1753]:
```suggestion
    "name": { "str": "Gottweiler" },
    "description": "This Rottweiler has grown to the proportions of a cow.  Its short fur is riddled with scars and the hide underneath has grown thick and leathery.",
```

File: data/json/monsters/mammal.json (hunk 1 above)
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json]:
```suggestion
    "name": { "str": "Gottweiler puppy", "str_pl": "gottweiler puppies" },
    "description": "An adorable, defenseless Gottweiler puppy.  Much safer to tame than an adult dog, in spite of its already impressive size.",
```

**Comment by Karol1223** [comment]
> Do you think a "PLAYER_CLOSE" anger flag for the Greatest Swiss would be a good idea in that case? Or is the "PLAYER_NEAR_BABY" flag enough?

It depends what you want to do with it. If I understand correctly and this is indeed the mutated version then absolutely yes - I'd honestly just make them aggressive at that point. You're looking at a predator larger than the native black bears. You're just food to them.

**Comment by f-rovers** [comment]
> > Do you think a "PLAYER_CLOSE" anger flag for the Greatest Swiss would be a good idea in that case? Or is the "PLAYER_NEAR_BABY" flag enough?
> 
> It depends what you want to do with it. If I understand correctly and this is indeed the mutated version then absolutely yes - I'd honestly just make them aggressive at that point. You're looking at a predator larger than the native black bears. You're just food to them.
> 
> Also while we're at it - you should move the mutants to the mutant mammal/wildlife/whatever it's called file as opposed to throwing them into the mammal file. Check the file that Cerbearus and Spideer are in.

I'll move them in a bit. Great tusked moose is also still in mammals iirc, move that one as well?

**Comment by Karol1223** [comment]
> Great tusked moose is also still in mammals iirc, move that one as well?

Moving existing code between files can mess with git blame, so that's up to you. If it's just one (I think two since it has a calf?) monster it shouldn't be a big issue to move it, though.

File: data/json/monsters/mutant_mammal.json [hunk 7]
```diff
@@ -379,6 +379,87 @@
...
+    "dissect": "dissect_cattle_sample_large",
+    "families": [ "prof_intro_biology" ],
+    "biosignature": { "biosig_item": "feces_manure", "biosig_timer": 8 },
+    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM" ]
+  },
+  {
+    "id": "mon_tusked_moose_calf",
+    "type": "MONSTER",
+    "name": { "str": "tusked moose calf", "str_pl": "tusked moose calves" },
+    "description": "The baby moose is called a calf so are it's tusked mutant descendants.  Calves can outrun a human within days of birth and gain weight at an incredible pace.",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:427]:
```suggestion
    "description": "The baby moose is called a calf so are its tusked mutant descendants.  Calves can outrun a human within days of birth and gain weight at an incredible pace.",
```

**Comment by f-rovers** [comment]
Just a thought. Now that the dogs are way harder to tame, we wouldn't need to err on the side of caution in terms of their power level. And admittedly, both of the evolved dogs aren't *that* dangerous. The 'midgame' preset character with 8/8/8/8 stats was handily able to beat either one on one in melee. On the other hand, I scaled up the stats in a semi-handwavy way, keeping the original breeds stats in mind. Especially the Greatest Swiss is kind of easy to beat for a horse sized hound.

**Comment by GuardianDll** [comment]
Why closed?

**Comment by f-rovers** [comment]
I"ve messed up with Github desktop by the looks of things, trying to fix it.

**Pull request closed without merging**